BigImageViewer/src/main/java/com/github/piasy/biv/view/BigImageView.java

```diff
@@ -317,19 +325,61 @@ public void saveImageIntoGallery() {
         }
 
         try {
-            String result = MediaStore.Images.Media.insertImage(getContext().getContentResolver(),
-                    mCurrentImageFile.getAbsolutePath(), mCurrentImageFile.getName(), "");
-            if (mImageSaveCallback != null) {
-                if (!TextUtils.isEmpty(result)) {
-                    mImageSaveCallback.onSuccess(result);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                ContentResolver resolver = getContext().getContentResolver();
+                ContentValues contentValues = new ContentValues();
+                contentValues.put(MediaStore.MediaColumns.DISPLAY_NAME, mCurrentImageFile.getName() + ".jpg");
+                contentValues.put(MediaStore.MediaColumns.MIME_TYPE, "image/jpg");
+                contentValues.put(MediaStore.MediaColumns.RELATIVE_PATH, Environment.DIRECTORY_PICTURES);
+                Uri imageUri = resolver.insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, contentValues);
+
+                if (imageUri != null) {
+                    OutputStream outputStream = resolver.openOutputStream(imageUri);
+                    BitmapFactory.Options options = new BitmapFactory.Options();
+                    options.inPreferredConfig = Bitmap.Config.ARGB_8888;
+                    Bitmap bitmap = BitmapFactory.decodeFile(mCurrentImageFile.getAbsolutePath(), options);
+
+                    boolean saved = bitmap.compress(Bitmap.CompressFormat.PNG, 100, outputStream);
+                    if (mImageSaveCallback != null) {
+                        if (saved) {
+                            mImageSaveCallback.onSuccess(imageUri.toString());
+                        } else {
+                            mImageSaveCallback.onFail(new RuntimeException("saveImageIntoGallery fail"));
+                        }
+                    }
+
+                    if (outputStream != null) {
```

Comment:
Should we flush and close it before firing callback?

Reply:
Yes, you are right, I correct it.